test/main.cpp

```diff
@@ -101,6 +101,26 @@ TEST_CASE("Ringbuffer wraps around on overflow", "[ringbuffer]") {
     REQUIRE(buffer.copy_and_pop() == 18);
 }
 
+TEST_CASE("Ringbuffer copy and pop front rolover correct", "[ringbuffer]") {
+    ringbuffer_c<int, 2> buffer;
+
+    // move tail to end of ringbuffer
+    buffer.push(5);
+    buffer.push(92);
```

Comment:
These look like magic numbers to me since you push them without rechecking for them